antiweb.py

```diff
@@ -1465,18 +1465,156 @@ def filter_output(self, lines):
                     #see the l.text.lstrip()! if the lines ends with a white space
                     #the quotes will be kept! This is feature, to force the quotes
                     #in the output
                     continue
                 
-                if stext.startswith("//") and not stext.startswith("#####"):
-                    #remove comments but not chapters
+                if stext.startswith("//"):
                     l.text = l.indented(stext[2:])
                             
             yield l
+#edoc
+
+#@cstart(CSharpReader)
+class CSharpReader(CReader):
+    #@start(CSharpReader doc)
+    #CSharpReader
+    #=============
+    """
+    .. py:class:: CSharpReader
+
+       A reader for C# code. This class inherits :py:class:`CReader`.
+    """
+    
+    #@indent 3
+    #@include(CSharpReader)
+    #@include(CSharpReader.strip_tags doc)
+    #@include(CSharpReader.get_attribute_text doc)
+    #@include(CSharpReader.filter_output doc)
+    #@(CSharpReader doc)
+    
+    def _accept_token(self, token):
+        return super(CSharpReader, self)._accept_token(token)
+    
+    def _cut_comment(self, index, token, text):
+        return super(CSharpReader, self)._cut_comment(index, token, text)
+        
+    #@cstart(CSharpReader.strip_tags)
+    def strip_tags(self,  soup):
+        """
+        .. py:method:: strip_tags(soup)
+
+           Removes all C# XML tags. The tags are replaced by their attributes and contents.
+
+           :param soup: The parsed xml tags.
+           :return: the XML tags replaced by their attributes and contents.
+        """
+        if isinstance(soup,  Tag) and not soup.contents:            
+            return self.get_attribute_text(soup)
+    
+        for tag in soup.find_all(True):
+            text = self.get_attribute_text(tag)
+            
+            #if the content is a Navigablestring the content is simply concatenated 
+            #otherwise the contents of the content are recursively checked for xml tags
+            for content in tag.contents:
+                if not isinstance(content, NavigableString):
+                    content = self.strip_tags(content)
+                text += content
+    
+            tag.replaceWith(text)
+    
+        return soup
+    #@edoc
+    
+    #@cstart(CSharpReader.get_attribute_text)
+    def get_attribute_text(self,  tag):
+        """
+        .. py:method:: get_attribute_text(tag)
+
+           Gets the values of all attributes of a XML tag.
+
+           :param tag: A BeautifulSoup Tag.
+           :return: the values of all attributes concatenated
+        """
+        #collect all values of xml tag attributes
+        attributes = tag.attrs
+        attribute_text = ""
+        for attribute, value in attributes.items():
+            attribute_text = value + " "
+        return attribute_text
+    #@edoc
+    
+    #@cstart(CSharpReader.filter_output)
+    def filter_output(self, lines):
```

Comment:
This is a rather long method. Could you 'antiwebify' it so that it tells a nice story in the documentation?